File: app/src/main/java/mchehab/com/knearestneighbor/distanceAlgorithm/ManhattenDistance.java
package mchehab.com.knearestneighbor.distanceAlgorithm;

public class ManhattenDistance implements DistanceAlgorithm {
    @Override
    public double calculateDistance(double x1, double y1, double x2, double y2) {
        double x = Math.abs(x1 - x2);
        double y = Math.abs(y1 - y2);
        double distance = x + y;
        return distance;
    }
}

File: kotlin/src/main/java/mchehab/com/kotlin/distanceAlgorithm/ManhattenDistance.kt
package mchehab.com.kotlin.distanceAlgorithm

import kotlin.math.abs

class ManhattenDistance: DistanceAlgorithm {
    override fun calculateDistance(x1: Double, x2: Double, y1: Double, y2: Double): Double {
        val x = abs(x1 - x2)
        val y = abs(y1 - y2)
        val distance = x + y
        return distance
    }
}